CONTRIBUTING.md

```diff
@@ -156,24 +161,54 @@ Powered by the [WindowsAppSDK](https://learn.microsoft.com/en-us/windows/apps/wi
 
 #### Developing on Windows
 
+##### WinUI
+
 Recommended IDE:
 - Visual Studio 2019 or Visual Studio 2022 with the required workloads (including .NET 7) and components for Windows app development. For more information, see [Install tools for the Windows App SDK](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/set-up-your-development-environment?tabs=cs-vs-community%2Ccpp-vs-community%2Cvs-2022-17-1-a%2Cvs-2022-17-1-b)
 
 Although, any IDE that supports .NET 7 and WindowsAppSDK should work.
 
 *Rider does support .NET 7, but has no WindowsAppSDK support at the time of writing this. Therefore, you will be able to develop part of the application use Rider, but will be unable to build and run the Windows platform version.*
 
+##### GNOME
+
+Follow developing on Linux instructions below to build the app in WSL. Make sure you use a distro in which you can install all required dependencies (OpenSUSE Tumbleweed is recommended).
+
 #### Developing on Linux
 
-Recommended IDE:
-- Builder 43 and up.
+##### WinUI
 
-You may also make your changes via any code editor and use `flatpak-builder` to run the application locally through flatpak.
+Can't be built on Linux.
+
+##### GNOME
 
-You may also simply install .NET 7 locally and use dotnet's CLI to run the application.
+Recommended IDEs:
+- GNOME Builder 43 and up.
+- VS Code with [flatpak extension](https://github.com/bilelmoussaoui/flatpak-vscode).
+
+You may also make your changes via any code editor and use `flatpak-builder` to run the application locally through flatpak.
 
-*Running through dotnet CLI won't give you all the desktop integration features provided when using Builder or `flatpak-builder`. 
-For example, resources will not be loaded and css files/custom icon files will not be displayed when running the app through cli.*
+You may also build the app manually without using flatpak. List of dependencies:
+- dotnet >=7.0
+- GTK >=4.10
+- libadwaita >=1.2
+- python3 >=3.7
+- yt-dlp (as python module, standalone binary won't work)
+- ffmpeg
+- aria2 (optional)
```

Comment:
Shouldn't these not be optional as if the setting is turned the app will crash as it can't find it?

Reply:
It will not crash, yt-dlp falls back to native downloader if it can't find aria2